apps/site/lib/site/realtime_schedule.ex

```diff
@@ -0,0 +1,222 @@
+
+    build_output(stops, route_with_patterns, schedules, predictions)
+  end
+
+  @spec get_stops([Stop.id_t()], fun()) :: map
+  defp get_stops(stop_ids, stops_fn) do
+    stop_ids
+    |> Enum.map(
+      &Task.async(fn ->
+        {&1, &1 |> stops_fn.() |> Map.take([:id, :name, :accessibility, :address])}
+      end)
+    )
+    |> Enum.into(%{}, &Task.await/1)
+  end
+
+  @spec get_routes([Stop.id_t()], fun()) :: [route_with_patterns_t]
+  defp get_routes(stop_ids, routes_fn) do
+    stop_ids
+    |> Enum.map(
+      &Task.async(fn ->
+        &1
+        |> routes_fn.()
+        |> Enum.map(fn {route, route_patterns} ->
+          {&1, Route.to_json_safe(route), route_patterns}
+        end)
+      end)
+    )
+    |> Enum.map(&Task.await/1)
+    |> List.flatten()
+  end
+
+  @spec get_predictions([route_with_patterns_t], DateTime.t(), fun()) :: map
+  defp get_predictions(route_with_patterns, now, predictions_fn) do
+    route_with_patterns
+    |> Enum.map(fn {stop_id, _route, route_patterns} ->
+      Task.async(fn ->
+        do_get_predictions(stop_id, route_patterns, now, predictions_fn)
+      end)
+    end)
+    |> Enum.flat_map(&Task.await/1)
+    |> Enum.into(%{})
+  end
+
+  @spec do_get_predictions(Stop.id_t(), [RoutePattern.t()], DateTime.t(), fun()) :: [
+          {
+            route_pattern_name_t,
+            [Prediction.t()]
+          }
+        ]
+  defp do_get_predictions(stop_id, route_patterns, now, predictions_fn) do
+    route_patterns
+    |> Enum.map(fn route_pattern ->
+      Task.async(fn ->
+        next_two_predictions =
+          [
+            stop: stop_id,
+            route_pattern: route_pattern.id,
+            min_time: now,
+            sort: "time",
+            "page[limit]": 2
+          ]
+          |> predictions_fn.()
+
+        {route_pattern.name, next_two_predictions}
+      end)
+    end)
+    |> Enum.map(&Task.await/1)
+  end
+
+  @spec get_schedules([route_with_patterns_t], DateTime.t(), fun()) :: map
+  defp get_schedules(route_with_patterns, now, schedules_fn) do
+    route_with_patterns
+    |> Enum.map(fn {stop_id, route, route_patterns} ->
+      Task.async(fn ->
+        do_get_schedules(route.id, stop_id, route_patterns, now, schedules_fn)
+      end)
+    end)
+    |> Enum.flat_map(&Task.await(&1, @long_timeout))
+    |> Enum.into(%{})
+  end
+
+  @spec do_get_schedules(String.t(), Stop.id_t(), [route_with_patterns_t], DateTime.t(), fun()) ::
+          map
+  defp do_get_schedules(route_id, stop_id, route_patterns, now, schedules_fn) do
+    route_pattern_dictionary = make_route_pattern_dictionary(route_patterns)
+
+    [route_id]
+    |> schedules_fn.(min_time: now)
+    |> Enum.filter(&(&1.stop.id == stop_id))
+    |> Enum.group_by(& &1.trip.route_pattern_id)
+    |> Enum.into(
+      %{},
+      fn {route_pattern_id, schedules} ->
+        {Map.get(route_pattern_dictionary, route_pattern_id), Enum.take(schedules, 2)}
```

Comment:
this `2` should also be in a module attribute, possibly the same one as above if it's semantically the same